api/datadoghq/v2alpha1/datadogpodautoscaler_types.go

```diff
@@ -0,0 +1,560 @@
+	// +kubebuilder:validation:Minimum=0
+	Value int32 `json:"value"`
+
+	// PeriodSeconds specifies the window of time for which the policy should hold true.
+	// PeriodSeconds must be greater than zero and less than or equal to 1800 (30 min).
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=1800
+	PeriodSeconds int32 `json:"periodSeconds"`
+}
+
+// DatadogPodAutoscalerTargetType defines the type of the target.
+// +kubebuilder:validation:Enum:=PodResource;ContainerResource
+type DatadogPodAutoscalerTargetType string
+
+const (
+	// DatadogPodAutoscalerResourceTargetType allows to set POD-level resources targets.
+	DatadogPodAutoscalerResourceTargetType DatadogPodAutoscalerTargetType = "PodResource"
+
+	// DatadogPodAutoscalerContainerResourceTargetType allows to set container-level resources targets.
+	DatadogPodAutoscalerContainerResourceTargetType DatadogPodAutoscalerTargetType = "ContainerResource"
+)
+
+// DatadogPodAutoscalerObjective defines the objectives to reach and maintain for the target resource.
+type DatadogPodAutoscalerObjective struct {
+	// Type sets the type of the target.
+	Type DatadogPodAutoscalerTargetType `json:"type"`
+
+	// PodResource allows to set a POD-level resource target.
+	PodResource *DatadogPodAutoscalerResourceTarget `json:"podResource,omitempty"`
+
+	// ContainerResource allows to set a container-level resource target.
+	ContainerResource *DatadogPodAutoscalerContainerResourceTarget `json:"containerResource,omitempty"`
+}
+
+// DatadogPodAutoscalerResourceTarget defines a POD-level resource target (for instance, CPU Utilization at 80%)
+// For POD-level targets, resources are the sum of all containers resources.
+// Utilization is computed from sum(usage) / sum(requests).
+type DatadogPodAutoscalerResourceTarget struct {
+	// Name is the name of the resource.
+	// +kubebuilder:validation:Enum:=cpu
+	Name corev1.ResourceName `json:"name"`
+
+	// Value is the value of the target.
+	Value DatadogPodAutoscalerTargetValue `json:"value"`
+}
+
+// DatadogPodAutoscalerContainerResourceTarget defines a Container level resource target (for instance, CPU Utilization for container named "foo" at 80%)
+type DatadogPodAutoscalerContainerResourceTarget struct {
+	// Name is the name of the resource.
+	// +kubebuilder:validation:Enum:=cpu
+	Name corev1.ResourceName `json:"name"`
+
+	// Value is the value of the target.
+	Value DatadogPodAutoscalerTargetValue `json:"value"`
+
+	// Container is the name of the container.
+	Container string `json:"container"`
+}
+
+// DatadogPodAutoscalerTargetValue defines the value of the target.
+type DatadogPodAutoscalerTargetValue struct {
+	// Type specifies how the value is expressed (Absolute or Utilization).
+	Type DatadogPodAutoscalerTargetValueType `json:"type"`
+
+	// Absolute defines the absolute value of the target (for instance 500 millicores).
+	Absolute *resource.Quantity `json:"absolute,omitempty"`
```

Comment:
Note: we need to keep Absolute for future cases but we need to disallow it for multi-dim and vertical scaling.

Reply:
I assume we will allow Absolute target for horizontal-only scaling from start. Am I correct?

If not then I'd remove this (since its never allowed) and add it later, when we start supporting this

Reply:
Removed. I think having unsupported fields is very confusing UX